### PR TITLE
Port changes from IBM/charts/pull/309

### PR DESCRIPTION
**Pull request by leszko**
Port changes from the PR sent to IBM Helm Charts: https://github.com/IBM/charts/pull/309

CC: @eminn (you may want to port some of these changes to Jet Helm Charts)

### CODE REVIEW

File: stable/hazelcast-enterprise/templates/mancenter-statefulset.yaml [hunk 1]
```diff
@@ -4,22 +4,25 @@ kind: Deployment
...
       role: mancenter
   template:
     metadata:
       labels:
-        app: {{ template "hazelcast.name" . }}
-        release: "{{ .Release.Name }}"
+        app.kubernetes.io/name: {{ template "hazelcast.name" . }}
+        helm.sh/chart: {{ template "hazelcast.chart" . }}
+        app.kubernetes.io/instance: "{{ .Release.Name }}"
+        app.kubernetes.io/managed-by: "{{ .Release.Service }}"
```
Comment (mesutcelik) [inline stable/hazelcast-enterprise/templates/mancenter-statefulset.yaml]: What are those labels? They are everywhere 😄

File: stable/hazelcast-enterprise/templates/tests/test-hazelcast.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+apiVersion: v1
+kind: Pod
```
Comment (mesutcelik) [inline stable/hazelcast-enterprise/templates/tests/test-hazelcast.yaml]: Are we going to run those tests before releasing new helm charts?

File: stable/hazelcast-enterprise/templates/tests/test-hazelcast.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    - "-c"
+    - |
+      set -ex
+      # Get the number of Hazelcast members in the cluster
+      CLUSTER_SIZE=$(curl {{ template "hazelcast.fullname" . }}:{{ .Values.service.port }}/hazelcast/health/cluster-size)
+      # Test the currect number of Hazelcast members
+      test ${CLUSTER_SIZE} -eq {{ .Values.cluster.memberCount }}
+    securityContext:
+      runAsNonRoot: true
+      runAsUser: 1001
```
Comment (mesutcelik) [inline stable/hazelcast-enterprise/templates/tests/test-hazelcast.yaml]: What is the difference between those securityContext parameters and the ones above?

File: stable/hazelcast-enterprise/templates/mancenter-statefulset.yaml (hunk 1 above)
Comment (leszko) [reply stable/hazelcast-enterprise/templates/mancenter-statefulset.yaml]: These are standard Helm Chart labels. There were before, however the naming convention changed,, so I had to rename them all.

These are the labels added to all resources created with the given Helm Chart.

File: stable/hazelcast-enterprise/templates/tests/test-hazelcast.yaml (hunk 2 above)
Comment (leszko) [reply stable/hazelcast-enterprise/templates/tests/test-hazelcast.yaml]: yeah, I'd like to. I already spoke to @Holmistr . We should set up the Kubernetes testing environment and add it to the pipeline. 

Jiri suggested that @hasancelik could do it when he's back :)

File: stable/hazelcast-enterprise/templates/tests/test-hazelcast.yaml (hunk 3 above)
Comment (leszko) [reply stable/hazelcast-enterprise/templates/tests/test-hazelcast.yaml]: these are for the container, the others are for the Pod. Apparently they may differ ;) It's also confusing to me...

**Pull request merged**